lib/facter/alert_manager_running.rb

```diff
@@ -0,0 +1,6 @@
+Facter.add('prometheus_alert_manager_running') do
+  setcode do
+    svc_status = Facter::Core::Execution.exec('puppet resource service alert_manager | grep ensure')
```

Comment:
do you have experience with calling puppet in a fact? Are there situations where it could deadlock? I am a bit scared that this will slow down all puppet run on each system in a environment.

Reply:
This is just puppet resource, not puppet apply or puppet agent. I tested the fact on some systems without any difficulties. I prefer to use puppet resource service as this is generic to all platforms. We can of course confine this to only run on linux platforms.
I am happy to accept any other, more smart idea.
Besides this: I was about to change this on Monday to puppet resource --yaml service ... to allow smarter data parsing.